plugins/askrene/reserve.c

```diff
@@ -83,6 +83,41 @@ static bool remove(struct reserve *r, struct amount_msat amount)
 	return true;
 }
 
+/* Just like reserves_add, but used to rewind a failed reserves_remove. It
+ * cannot fail cause we don't want recursion. */
+static void reserves_add_no_fail(struct reserve_htable *reserved,
+				 const struct short_channel_id_dir *scidds,
+				 const struct amount_msat *amounts, size_t num)
+{
+	bool success;
+	for (size_t i = 0; i < num; i++) {
+		struct reserve *r = reserve_htable_get(reserved, &scidds[i]);
+		if (!r)
+			r = new_reserve(reserved, &scidds[i]);
+		assert(r);
+		success = add(r, amounts[i]);
+		assert(success);
+	}
+}
+
+/* Just like reserves_remove, but used to rewind a failed reserves_add. It
+ * cannot fail cause we don't want recursion. */
+static void reserves_remove_no_fail(struct reserve_htable *reserved,
+				    const struct short_channel_id_dir *scidds,
+				    const struct amount_msat *amounts,
+				    size_t num)
+{
+	bool success;
+	for (size_t i = 0; i < num; i++) {
+		struct reserve *r = reserve_htable_get(reserved, &scidds[i]);
+		assert(r);
```

Comment:
if there's no `r` it's fine, you can just continue right? You're trying to remove it?

Reply:
Yes I am trying to remove it. It should be there because I only call this function to remove the reserves that I already added. If the reserve isn't there it means there is a bug somewhere.

I see that one could skip it and pretend nothing is wrong... IDK